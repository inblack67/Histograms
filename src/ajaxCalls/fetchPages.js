import axios from 'axios';

export default async () => {
    try {
        return await axios('https://www.vdocipher.com/blog/wp-json/wp/v2/pages?per_page=100');
    } catch (err) {
        throw new Error(`Request failed with a status code of ${err.response.status}`);
    }
}
