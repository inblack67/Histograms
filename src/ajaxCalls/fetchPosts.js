import axios from 'axios';

export default async () => {
    try {
        return await axios('https://www.vdocipher.com/blog/wp-json/wp/v2/posts?per_page=100');
    } catch (err) {
        throw new Error(`Request failed with a status code of ${err.response.status}`);
    }
}
