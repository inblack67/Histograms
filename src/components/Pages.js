import React, { useEffect, useState } from 'react';
import HighCharts from 'highcharts';
import ReactHighCharts from 'highcharts-react-official';
import fetchPages from '../ajaxCalls/fetchPages';

const Posts = () => {

    const [pages, setPages] = useState([]);
    const [loading, setLoading] = useState(true);
    const wordCount = [];

    useEffect(() => {
        getPages();
    },[])

    const getPages = async () => {
        const res = await fetchPages();
        setPages(res.data);
        setLoading(false);
    }

    const getWordCount = () => {
        if(pages){
            const res = pages.map(page => page.content.rendered.match(/\S+/gi));
            res.forEach(r => {
                if(r){
                    wordCount.push(r.length)
                }
            })
        }
    }

    getWordCount();

    const options = {
        chart: {
            type: 'column',
            zoomType: 'xy'
        },
        title: {
            text: 'Pages Word Count',
            style: {
                color: 'tomato'
            }
        },
        tooltip: {
            shared: true,
            formatter(){
                return `<strong>Pages: ${this.x}</strong>
                <br /><strong>Words: ${this.y}</strong>`
            }
        },
        yAxis: {
            title: {
                text: 'Words'
            },
        },
        xAxis: {
            title: {
                text: 'Pages'
            },
        },
        credits: {
            enabled: false
        },
        colors: ['tomato'],
        series: [
            {
                name: 'Pages',
                data: wordCount
            }
        ]
    }

    if(loading){
        return <h5 className='text-center'>Loading...</h5>
    }

    return (
        <div className="container">
            <ReactHighCharts options={options} highcharts={HighCharts} />
        </div>
    )
}

export default Posts
