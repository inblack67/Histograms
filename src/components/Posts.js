import React, { useEffect, useState } from 'react';
import HighCharts from 'highcharts';
import ReactHighCharts from 'highcharts-react-official';
import fetchPosts from '../ajaxCalls/fetchPosts';

const Posts = () => {

    const [posts, setPosts] = useState([]);
    const [loading, setLoading] = useState(true);
    const wordCount = [];

    useEffect(() => {
        getPosts();
    },[])

    const getPosts = async () => {
        try {
            const res = await fetchPosts();
            setPosts(res.data);
            setLoading(false);
        } catch (err) {
            console.error(err);
        }
    }

    const getWordCount = () => {
        if(posts){
            const res = posts.map(post => post.content.rendered.match(/\S+/gi));
            res.forEach(r => {
                if(r){
                    wordCount.push(r.length)
                }
            })
        }
    }

    getWordCount();

    const options = {
        chart: {
            type: 'column',
            zoomType: 'xy'
        },
        title: {
            text: 'Posts Word Count',
            style: {
                color: 'blue'
            }
        },
        tooltip: {
            shared: true,
            formatter(){
                return `<strong>Posts: ${this.x}</strong>
                <br /><strong>Words: ${this.y}</strong>`
            }
        },
        yAxis: {
            title: {
                text: 'Words'
            },
        },
        xAxis: {
            title: {
                text: 'Posts'
            },
        },
        credits: {
            enabled: false
        },
        series: [
            {
                name: 'Posts',
                data: wordCount
            }
        ]
    }

    if(loading){
        return <h5 className='text-center'>Loading...</h5>
    }

    return (
        <div className="container">
            <ReactHighCharts options={options} highcharts={HighCharts} />
        </div>
    )
}

export default Posts
