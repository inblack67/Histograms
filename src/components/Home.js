import React, { Fragment, useState } from 'react'
import Posts from './Posts';
import Pages from './Pages';

const Home = () => {

    const [clicked, setClick] = useState(false);

    const onClick = e => {
      setClick(true);
    }

    return (
        <Fragment>

            { !clicked &&  <div className='btn-container'>
            <a href="#!" className='btn' onClick={onClick} data-testid='clicker'>
                Load Data
            </a>
        </div> }

            { clicked && <Fragment>
            <p className='text-center' data-testid='isloaded'>
                Data Loaded: true
            </p>
            <Posts />
            <hr/>
            <Pages />
            </Fragment> }
        </Fragment>
    )
}

export default Home
