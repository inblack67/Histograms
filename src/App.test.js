import React from 'react';
import { render, fireEvent, cleanup } from '@testing-library/react';
import fetchPosts from './ajaxCalls/fetchPosts';
import fetchPages from './ajaxCalls/fetchPages';
import Pages from './components/Pages'
import Home from './components/Home'
import Posts from './components/Posts'

afterEach(cleanup);

// renders
it('renders posts histogram correctly', () => {
  const { asFragment } = render(<Posts />)
  expect(asFragment).toMatchSnapshot();

})

it('renders pages histogram correctly', () => {
  const { asFragment } = render(<Pages />)
  expect(asFragment).toMatchSnapshot();

})

// async
it('fetch posts', async () => {
  const res = await fetchPosts();
  expect(res.data[0].categories).toEqual([10,56,77]);
})

it('fetch pages', async () => {
  const res = await fetchPages();
  expect(res.data[0].categories).toEqual([10,56,77]);
})

// events
it('state changes on click', () => {
  const { getByTestId } = render(<Home />);

  fireEvent.click(getByTestId('clicker'));

  expect(getByTestId('isloaded')).toHaveTextContent('Data Loaded: true');
})


