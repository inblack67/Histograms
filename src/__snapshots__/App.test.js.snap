// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders pages histogram correctly 1`] = `[Function]`;

exports[`renders posts histogram correctly 1`] = `[Function]`;
